### ISSUE TITLE
`--help` inconsistency in the CLI

### ISSUE THREAD

**Issue by salaheldinsoliman** [iotaledger/iota#4040]
## Description

`--help` displays every flag available for every subcommand. That is, if running `iota move --help`, it will display:

![Image](https://github.com/user-attachments/assets/83c6cb58-18f4-4f52-bf13-18f4c8f7b5d6)

That is the same output of `iota move test --help` for instance, which is really confusing.
In summary, flags shown should be only related to the subcommand in question

Related to #1417

**Comment by thibault-martinez** [comment]
Hi @salaheldinsoliman, did you make any progress on this one? Let us know if you need help.

**Comment by salaheldinsoliman** [comment]
Hi @thibault-martinez , I will start working on it now, will let you know If I need any help 👍

**Comment by salaheldinsoliman** [comment]
@thibault-martinez It seems that the arguments of [`BuildConfig`](https://github.com/iotaledger/iota/blob/9b9b39ed77df1dc99fb5f4e95a8a0c58e4d74c21/external-crates/move/crates/move-package/src/lib.rs#L49)
are used in every subcommand of `iota move`. So, this actually not an inconsistency except for `iota move new` which takes only a single argument and doesn't require re-building the package.
I suggest leaving this as is, WDYT?

**Comment by thibault-martinez** [comment]
Yeah I just had a look as well and it was definitely on purpose, closing this. Thanks.
